frontend/packages/kubevirt-plugin/src/components/vms/pending-changes-warning.tsx

```diff
@@ -0,0 +1,200 @@
+  <Breadcrumb className="kv-warning--pf-c-breadcrumb">
+    <BreadcrumbHeading key={`${tabName}-header-${tabProps.join('-')}`}>{tabName}</BreadcrumbHeading>
+    <BreadcrumbItem key={`${tabName}-${tabProps.join('-')}`}>
+      {tabProps.map((key, idx) => (
+        <div key={`${tabName}-${key}`} className="kv-warning-changed-attrs">
+          <Button isInline onClick={pendingChanges[key].execAction} variant={ButtonVariant.link}>
+            {key}
+          </Button>
+          <div className="kv-warning--comma-margin">{idx === tabProps.length - 1 ? '' : ','}</div>
+        </div>
+      ))}
+    </BreadcrumbItem>
+  </Breadcrumb>
+);
+
+export const getPendingChanges = (
+  vm: VMKind,
+  vmWrapper: VMWrapper,
+  vmiWrapper: VMIWrapper,
+): PendingChanges => ({
+  [IsPendingChange.flavor]: {
+    isPendingChange: isFlavorChanged(vmWrapper, vmiWrapper),
+    execAction: () => vmFlavorModal({ vmLike: vm, blocking: true }),
+    vmTab: VMTabEnum.details,
+  },
+  [IsPendingChange.cdroms]: {
+    isPendingChange: isCDROMChanged(vmWrapper, vmiWrapper),
+    execAction: () => VMCDRomModal({ vmLikeEntity: vm, modalClassName: 'modal-lg' }),
+    vmTab: VMTabEnum.details,
+  },
+  [IsPendingChange.bootOrder]: {
+    isPendingChange: isBootOrderChanged(vmWrapper, vmiWrapper),
+    execAction: () => BootOrderModal({ vmLikeEntity: vm, modalClassName: 'modal-lg' }),
+    vmTab: VMTabEnum.details,
+  },
+  [IsPendingChange.env]: {
+    isPendingChange: isEnvDisksChanged(vmWrapper, vmiWrapper),
+    execAction: () =>
+      history.push(`/ns/${getNamespace(vm)}/virtualmachines/${getName(vm)}/environment`),
+    vmTab: VMTabEnum.env,
+  },
+  [IsPendingChange.nics]: {
+    isPendingChange: isNicsChanged(vmWrapper, vmiWrapper),
+    execAction: () => history.push(`/ns/${getNamespace(vm)}/virtualmachines/${getName(vm)}/nics`),
+    vmTab: VMTabEnum.nics,
+  },
+  [IsPendingChange.disks]: {
+    isPendingChange: isDisksChanged(vmWrapper, vmiWrapper),
+    execAction: () => history.push(`/ns/${getNamespace(vm)}/virtualmachines/${getName(vm)}/disks`),
+    vmTab: VMTabEnum.disks,
+  },
+});
+
+export const PendingChangesWarning: React.FC<PendingChangesWarningProps> = ({
+  vmLikeEntity,
+  vmi: vmiProp,
+}) => {
+  const vm: VMKind = asVM(getLoadedData(vmLikeEntity));
+  const vmi = getLoadedData(vmiProp);
+
+  const vmWrapper = new VMWrapper(vm);
+  const vmiWrapper = new VMIWrapper(vmi);
+
+  const [pendingChanges, setPendingChanges] = React.useState<PendingChanges>(
+    getPendingChanges(vm, vmWrapper, vmiWrapper),
+  );
+
+  React.useEffect(() => {
+    setPendingChanges(getPendingChanges(vm, vmWrapper, vmiWrapper));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [vm, vmi]);
+
+  const keys = Object.keys(pendingChanges);
+  const arePendingChanges = keys.reduce(
+    (boolVal, key) => boolVal || pendingChanges[key].isPendingChange,
+    false,
+  );
+
+  if (!isVMRunningOrExpectedRunning(vm) || keys.length === 0 || !arePendingChanges) {
```

Comment:
isVMRunningOrExpectedRunning(vm) could be checked at the beggining

Reply:
it can't since the hooks can't be conditional

Reply:
if you remove the hook it can